### ISSUE TITLE
[Feature Request] Change inspection options for EventHandler function args

### ISSUE THREAD

**Issue by drewhoener** [DemonWav/MinecraftDevIntelliJ#675]
I think it'd be useful to be able to change the parameter inspections for Bukkit/Spigot/Paper eventhandler methods so that classes aren't marked as having errors when not needed.

### Context
I'm writing a minigame plugin that makes use of custom kits, and it's useful for me to write event handlers where I already have the player with the kit so I can fire off an event for each player with the kit.
So I wrote custom RegisteredListeners, making my event structure
```java
@EventHandler
public void onPotionThrow(Player kitPlayer, ProjectileLaunchEvent event){
    //Code specific to the player that has this specific kit
}
```
This makes MinecraftDev mark almost every one of my kit classes as having errors because the first parameter obviously isn't a subclass of `Event`

TL;DR Asking for an Inspection settings option for the plugin that allows downgrading the inspection to warning or soft warn so that they're not detected as real errors and flag the entire class

**Comment by RedNesto** [comment]
The thing is that the listeners analysis is made via an annotator, not an inspection, that means there is no way to disable it.

I see two solutions:
- keep using an annotator and add an option to disable this specific annotation for bukkit-based projects
- move to an inspection, which can be disabled and configured like any other in the IDE

It would also be possible to avoid these if you swap the two parameters, the plugin only checks the first parameter and ignores the other ones.
